File: 1.InvertireUnaStringa/invertireStringa.cpp
#include <iostream>
#include <string>

class Solution {
public:
    std::string reverseString(const std::string& str) {
        if (str.empty()) {
            return str;
        }
        
        
        return reverseString(str.substr(1)) + str[0];
    }
};

int main() {
    std::string str = "hello";
    std::cout << "Original: " << str << std::endl;

    Solution sol;
    std::string reversed = sol.reverseString(str);
    
    std::cout << "Reversed: " << reversed << std::endl;

    return 0;
}

File: 1.InvertireUnaStringa/invertireStringa.py
class Solution:
    def reverse_string(self, s: str) -> str:
        if not s:
            return s
        # Recursively call on the substring and add the first character after recursion
        return self.reverse_string(s[1:]) + s[0]

if __name__ == "__main__":
    s = "hello"
    print("Original:", s)
    
    sol = Solution()
    reversed_s = sol.reverse_string(s)
    
    print("Reversed:", reversed_s)

